src/bindingHandlers/copyToClipboard.ts

```diff
@@ -1,10 +1,13 @@
 import * as ko from "knockout";
 
 ko.bindingHandlers["copyToClipboard"] = {
-    init: (element: HTMLElement, valueAccessor: () => string): void => {
-        const copyToClipboard = () => {
+    init: (element: HTMLElement, valueAccessor: () => (string | (() => Promise<string>))): void => {
+        const copyToClipboard = async () => {
```

Comment:
If there is no `await` in the function, having `async` and `Promise` is not necessary in this case.

Reply:
There is an `await` for that promise that could be accepted as parameter. We need to accept it as Promise because `getPlainTextCodeSample` needs to await the `render` method from `TemplatingService`

Reply:
When the secrets are hidden, I replace each character of them with the character '•' in the `displayValue` property. When the user clicks copy, if he did not previously revealed the secrets, the `displayValue` will still be composed only from '•'. From what I understood, we don't want the user to copy the code template like this, but with the actual value. For this I have only found 2 solutions:
1. Keep two string rendered values one with the actual value and one with them replaced with '•' and always copy the one with plain values. But this means that we will have to render them both whenever a header changes.
2.  Only render the plain value when `copyToClipboard` is triggered and keep an already rendered value (depending on the `revealed` property) for the code sample.

Do you think the first option would be better? Or do you have another idea?

Reply:
Ok, I see now. Looks like your original variant is the most optimal in this case. Let's keep it then.